### ISSUE TITLE
[2.2.11] Crash when I open the sdk

### ISSUE THREAD

**Issue by matir91** [Tap-Payments/goSellSDK-Android#21]
This is the stacktrace , in the version [2.2.7]  it didn't happen
```
Fatal Exception: java.lang.NullPointerException: throw with null exception
       at company.tap.gosellapi.internal.api.responses.SDKSettings$Data.getPermissions(SDKSettings.java)
       at company.tap.gosellapi.internal.data_managers.payment_options.PaymentOptionsDataManager$ViewModelsHandler.generateCardPaymentModel + 993(PaymentOptionsDataManager.java:993)
       at company.tap.gosellapi.internal.data_managers.payment_options.PaymentOptionsDataManager$ViewModelsHandler.generateViewModels + 802(PaymentOptionsDataManager.java:802)
       at company.tap.gosellapi.internal.data_managers.payment_options.PaymentOptionsDataManager$ViewModelsHandler.access$100 + 732(PaymentOptionsDataManager.java:732)
       at company.tap.gosellapi.internal.data_managers.payment_options.PaymentOptionsDataManager.(PaymentOptionsDataManager.java:5)
       at company.tap.gosellapi.internal.data_managers.PaymentDataManager.createPaymentOptionsDataManager + 305(PaymentDataManager.java:305)
       at company.tap.gosellapi.internal.api.facade.GoSellAPI$1.onSuccess + 232(GoSellAPI.java:232)
       at company.tap.gosellapi.internal.api.facade.GoSellAPI$1.onSuccess + 228(GoSellAPI.java:228)
       at company.tap.gosellapi.internal.api.callbacks.BaseCallback.onResponse + 36(BaseCallback.java:36)
       at retrofit2.ExecutorCallAdapterFactory$ExecutorCallbackCall$1$1.run + 71(ExecutorCallAdapterFactory.java:71)
       at android.os.Handler.handleCallback + 754(Handler.java:754)
       at android.os.Handler.dispatchMessage + 95(Handler.java:95)
       at android.os.Looper.loop + 165(Looper.java:165)
       at android.app.ActivityThread.main + 6375(ActivityThread.java:6375)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run + 912(ZygoteInit.java:912)
       at com.android.internal.os.ZygoteInit.main + 802(ZygoteInit.java:802)
```

**Comment by Haitham001** [comment]
Could you please send me your secret key and app bundle (app package name)
to test it in my side.

*Sincerely,*
*Haitham Elsheshtawy*
*http://about.me/Haitham.Sheshtawy <http://about.me/Haitham.Sheshtawy>*

**Comment by matir91** [comment]
I checked and it's happening only if I have the proguard enabled.
This was the rule that I was using in [2.2.7] and it was working fine
```
-keep public class gotap.com.tapglkitandroid.** { *; }
```

**Comment by matir91** [comment]
This is a proguard issue, I fixed but #17 is not working property. You have to add the  `tapglkitandroid` rule and another for your models.

**Comment by HaithamSheshtawy** [comment]
Great. thank you very much.
